metals/src/main/scala/scala/meta/internal/metals/debug/server/DebugeeParamsCreator.scala

```diff
@@ -31,29 +39,42 @@ class DebugeeParamsCreator(buildTargets: BuildTargets) {
         .filter(_.nonEmpty)
         .getOrElse(Nil)
       val debugLibs = libraries.flatMap(createLibrary(_))
-      val includedInLibs = debugLibs.map(_.absolutePath).toSet
-
-      val classpath = buildTargets.targetJarClasspath(id).getOrElse(Nil)
-
-      val filteredClassPath = classpath.collect {
-        case path if !includedInLibs(path.toNIO) => UnmanagedEntry(path.toNIO)
-      }.toList
 
       val modules = buildTargets
         .buildTargetTransitiveDependencies(id)
         .flatMap(buildTargets.jvmTarget)
         .map(createModule(_))
         .toSeq
 
-      val scalaVersion = buildTargets.scalaTarget(id).map(_.scalaVersion)
+      val includedInLibsOrModules =
+        debugLibs.map(_.absolutePath).toSet ++ modules.map(_.absolutePath).toSet
+
+      for {
+        classpathString <- buildTargets
+          .targetClasspath(id, Promise())
```

Comment:
Could we forward cancelPromise from startDebugSession? This would allow us to cancel the classpath request when the user cancels debugging